### ISSUE TITLE
Escaping commas from CN

### ISSUE THREAD

**Issue by pbecotte** [jupyterhub/ldapauthenticator#139]
Spent a while trying to get this to work today. Turned out the issue is this snippet of code-

```python
        if self.lookup_dn:
            if str(self.lookup_dn_user_dn_attribute).upper() == "CN":
                # Only escape commas if the lookup attribute is CN
                username = re.subn(r"([^\\]),", r"\1\,", username)[0]
```

Was turning the username my AD returned `"Becotte, Paul"` into `"Becotte\, Paul"`, which fails auth. I am FAR from an AD expert and have no idea under what circumstances that escape may or may not be appropriate, but there was no combination of settings that would make my ldap auth bind with the search turned on. Fortunately, my setup appears to work fine with the search completely disabled (again, no idea if that is reliable or not!) but it feels like the `escape_dn_username` = False flag would also prevent the comma from being escaped?

(The latest version in this repo actually would have worked- I could have used attribute="uid", and a recent commit turns ['pbecotte'] into 'pbecotte' which would work, but there hasn't been a release since that change).

**Comment by Ownercz** [comment]
Can confirm, after commenting out this part of the code, everything works.

**Comment by consideRatio** [comment]
This was fixed by #267
